lib/commons/dom/get-element-stack.js

```diff
@@ -0,0 +1,489 @@
+/* global dom */
+
+// split the page cells to group elements by the position
+const gridSize = 200; // arbitrary size, increase to reduce memory (less cells) use but increase time (more nodes per grid to check collision)
+
+// https://developer.mozilla.org/en-US/docs/Web/CSS/Replaced_element
+const replacedElements = ['iframe', 'video', 'embed', 'img'];
```

Comment:
Should an image in the `:before` or `:after` of an element create a separate, replaced layer? Or is it considered the same as the host? What if the bounding rect is different?